server/lib/nl/scraper.py

```diff
@@ -238,88 +256,88 @@ def _check_svgs_drawn(driver):
     return _OVERRIDE_CHART_MAP[query.lower()]
 
-  logging.info(f'Scraping: {query}')
-  driver.get(_URL + query)
+  url = cfg.WEBSITE_ROOT + f'/nl/data#a=True&q={query}'
+  logging.info(f'Scraping: {url}')
+  driver.get(url)
 
   # Wait until the page has loaded.
   wait = WebDriverWait(driver, timeout=30, poll_frequency=0.25)
   wait.until(EC.presence_of_element_located((By.CLASS_NAME, 'chart-container')))
   wait.until(lambda d: _check_svgs_drawn(driver))
 
-  html = driver.page_source
-  soup = BeautifulSoup(html, 'html.parser')
-
   charts = []
-  subject_page = soup.find('div', {'id': 'subject-page-main-pane'})
-  for block in subject_page.find_all('section', {'class': 'block'}):
-    for chart_container in block.find_all('div', {'class': 'chart-container'}):
+  subject_page = driver.find_element(By.ID, 'subject-page-main-pane')
+  for block in subject_page.find_elements(By.CSS_SELECTOR, 'section.block'):
+    for chart_container in block.find_elements(By.CLASS_NAME,
+                                               'chart-container'):
       chart = {}
 
+      # Get title.
+      chart['title'] = chart_container.find_element(By.TAG_NAME, 'h4').text
+
+      # Get sources.
       chart['srcs'] = []
-      chart['title'] = chart_container.find('h4').text
-      for src in chart_container.find_all('div', {'class': 'sources'}):
-        anchor = src.find('a')
-        chart['srcs'].append({'name': anchor.text, 'url': anchor.get('href')})
-      classes = chart_container['class']
+      for src in chart_container.find_elements(By.CSS_SELECTOR, 'div.sources'):
+        anchor = src.find_element(By.TAG_NAME, 'a')
+        chart['srcs'].append({
+            'name': anchor.text,
+            'url': anchor.get_attribute('href')
+        })
+
+      # Get data csv.
+      tile_data = _get_tile_data(chart_container)
+      if tile_data:
+        data_csv = tile_data.get_attribute('data-csv')
+        if data_csv:
+          chart['data_csv'] = data_csv
+
+      classes = chart_container.get_attribute('class').split()
       if 'line-chart' in classes:
         chart['type'] = 'LINE'
-        chart['svg'] = _to_svg(chart_container.find('svg'))
-        legend = chart_container.find('div', {'class': 'legend'})
-        chart['legend'] = []
-        for l in legend.find_all('a'):
-          # TODO: maybe convert rbg to color name.
-          chart['legend'].append(l.text)
+        chart['svg'] = _get_inline_svg(chart_container)
+        _get_legend_data(chart_container, 'div.legend', 'a', chart)
       elif 'map-chart' in classes:
         chart['type'] = 'MAP'
-        map_area = chart_container.find('div', {'class': 'map'})
-        chart['svg'] = _to_svg(map_area.find('svg'))
-        legend_area = chart_container.find('div', {'class': 'legend'})
-        chart['legend_svg'] = _to_svg(legend_area.find('svg'))
+        map_area = chart_container.find_element(By.CSS_SELECTOR, 'div.map')
+        chart['svg'] = _get_inline_svg(map_area)
```

Comment:
map chart should be using data-svg now right?

Reply:
Just sync'd your PR.  Fixed.